extensions/most-used-content/app.R

```diff
@@ -0,0 +1,98 @@
+library(shiny)
+library(bslib)
+library(DT)
+library(connectapi)
+library(dplyr)
+library(purrr)
+library(lubridate)
+
+shinyOptions(
+  cache = cachem::cache_disk("./app_cache/cache/", max_age = 60 * 60 * 8)
+)
+
+source("get_usage.R")
+
+ui <- page_fillable(
+  theme = bs_theme(version = 5),
+
+  card(
+    card_header("Most Used Content"),
+    layout_sidebar(
+      sidebar = sidebar(
+        title = "No Filters Yet",
+        open = FALSE
+      ),
+      card(
+        DTOutput(
+          "content_usage_table"
+        )
+      )
+    )
+  )
+)
+
+server <- function(input, output, session) {
+  client <- connect()
+
+
+  # Default dates. "This week" is best "common sense" best represented by six
+  # days ago thru the end of today. Without these, content takes too long to
+  # display on some servers.
+  date_range <- reactive({
+    list(
+      from_date = today() - ddays(6),
+      to_date = today()
+    )
+  })
+
+  content <- reactive({
+    get_content(client)
+  }) |> bindCache("static_key")
+
+  usage_data <- reactive({
+    get_usage(
+      client,
+      from = format(date_range()$from_date, "%Y-%m-%dT%H:%M:%SZ"),
+      to = format(date_range()$to_date + hours(23) + minutes(59) + seconds(59), "%Y-%m-%dT%H:%M:%SZ")
+    )
+  }) |> bindCache(date_range()$from_date, date_range()$to_date)
```

Comment:
It feels wrong to format these dates here and not inside of `get_usage()`. Also, strictly speaking making this reactive now is extra complexity that is not needed right now. We know we will at some point do it, yes, but we should not take on the maintenance burden today for something we will do in the future. (Yes, in this sequence of PRs we expect to get to it quickly, but this is the same pattern that makes YAGNI so pernicious)

Reply:
Hmm, interesting. Yes, I can see how the formatting should take place inside the function. I was lumping the formatting and date math together, but I'll separate them.

The only reason it's reactive here is so that it can be cached (only reactives can be passed into `bindCache()`).

Reply:
I do wonder whether these should be in the time zone of the Connect server.

Reply:
> I do wonder whether these should be in the time zone of the Connect server.

This is going to end up being messy almost anyway you split it. Our API should either accept fully specified timestamps (with timezones) or only accept UTC (and our client code should convert to that form whatever is local) and then do what it needs to to make that work on the server (i.e. if the server is non-utc, then make sure the comparisons are timezone-aware and not comparing a UTC to some other timezone)